backend/sirius-web-emf/src/main/java/org/eclipse/sirius/web/emf/services/EditService.java

```diff
@@ -275,9 +298,13 @@ public void delete(Object object) {
         return createdObjectOptional;
     }
 
-    private Optional<EClass> getMatchingEClass(String namespaceId, String rootObjectCreationDescriptionId) {
+    private Optional<EClass> getMatchingEClass(UUID editingContextId, String namespaceId, String rootObjectCreationDescriptionId) {
+        EPackageRegistryImpl ePackageRegistry = new EPackageRegistryImpl(this.globalEPackageRegistry);
+        List<EPackage> additionalEPackages = this.editingContextEPackageService.getEPackages(editingContextId);
+        additionalEPackages.forEach(ePackage -> ePackageRegistry.put(ePackage.getNsURI(), ePackage));
```

Comment:
It's the 3rd time this segment is duplicated in this class; maybe extract in a properly named helper method.